### ISSUE TITLE
warn on `vault auth` if VAULT_TOKEN is set

### ISSUE THREAD

**Issue by mitchellh** [hashicorp/vault#195]
If you `vault auth` when a `VAULT_TOKEN` is set, you'll get some weird errors. We need to clarify that that token will take priority.

**Comment by EvanKrall** [comment]
This makes the [acl walkthrough](https://www.vaultproject.io/intro/getting-started/acl.html) really confusing:

```
juroujin:vault_test meatmanek$ vault token-create -policy=secret
4c5b313b-351b-9249-70f1-e7bc19ef58ca
juroujin:vault_test meatmanek$ vault auth 4c5b313b-351b-9249-70f1-e7bc19ef58ca
Successfully authenticated! The policies that are associated
with this token are listed below:

root
```

**Comment by sheldonh** [comment]
I just got bitten by this in the acl walkthrough. The work-around is:

```
VAULT_TOKEN="" vault auth 7bdcb522-0a09-55e8-ab17-a8f6f7d1ba15
```

I'll submit a doc PR.

**Comment by sheldonh** [comment]
I closed #267 in favour of #270, which prevents the dev mode server from recommending that VAULT_TOKEN be exported in the first place. If people just steer clear of that, all goes well.

But yes, even with that change in place, it would still be useful for `vault auth` to warn that it's using `VAULT_TOKEN` when it does so.

Would you consider it an error to specify arguments to `vault auth` when the token is present in the environment? Should arguments override the environment? Or is a warning message th e most sensible response?
